app/services/permissions_checker.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class PermissionsChecker
+  def initialize(current_user, permission_name, params)
+    @current_user = current_user
+    @permission = Permission.find_by(name: permission_name)
+    @params = params
+  end
+
+  def call
+    return true if RolePermission.joins(:permission).where(role_id: @current_user.role_id).find_by(permission_id: @permission.id).value == 'true'
+    return true if authorize_manage_user || authorize_manage_rooms || authorize_manage_recordings
+
+    false
+  end
+
+  private
+
+  def authorize_manage_user
```

Comment:
I don't think that this what we're going with,
APIs restriction will only be used for APIs scoped under `Admin` namespace.
All of the other, regular APIs will be rectified to work only on the authenticated user resources such as its rooms, recordings,... 
The regular APIs will expect authorize the user if  it's authenticated or not depending on the case.

With that being said all of these private methods functionality may not be essential in the upcoming weeks ,

Reply:
Also, the way of using the `params` as is, fitting all possible scenarios at once is a bit error prone,

Reply:
Isn't it what we are doing here?

Reply:
I'll double check but I think we are using regular endpoint for most things in the admin panel.

Reply:
I agree tho, I think they should be separate endpoints.

Reply:
I've been going through your comment a couple of time, and I think we have a different vision of how auth could/should be. Maybe you could provide an example for tomorrow's (5 aug) meeting?

Reply:
Sure, will sync in our meeting